bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -203,7 +203,6 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 				bootstrapv1.DataSecretAvailableCondition,
 				bootstrapv1.CertificatesAvailableCondition,
 			),
-			conditions.WithStepCounter(),
```

Comment:
Removing step counter from the KubeadmConfig object because it really does not makes sense for a process so simple (wrt to e.g. infrastructure provisioning)